### PR TITLE
Make care single source of truth for asset config on middleware

### PR DESCRIPTION
**Pull request by sainak**
## Proposed Changes

- add middleware level auth
- cache middleware openid keys
- cleanup middleware auth
- add route to fetch asset configs for middlewares
- add handlers to push config updates to middlewares
- allow middleware asset to access related assetbeds
- add a flag to indicate ocr log update status

### Associated Issue

- required for https://github.com/coronasafe/teleicu_middleware/issues/118
- partially addresses https://github.com/coronasafe/care/issues/2069

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Review by Ashesh3 — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Review by rithviknishad — APPROVED** [review]

File: care/facility/api/serializers/asset.py [hunk 1]
```diff
@@ -210,13 +216,55 @@ def validate(self, attrs):
...
+                meta.get("middleware_hostname")
+                or current_location.middleware_address
+                or current_location.facility.middleware_address
+            )
+            if ip_address and middleware_hostname:
+                assets_using_ip = (
+                    Asset.objects.filter(
+                        current_location__facility=current_location.facility
+                    )
+                    .annotate(
```
Comment (vigneshhari) [inline care/facility/api/serializers/asset.py]: Refactor

File: care/facility/api/serializers/asset.py [hunk 2]
```diff
@@ -271,11 +319,54 @@ def update(self, instance, validated_data):
                     asset=instance,
                     performed_by=user,
                 ).save()
-            updated_instance = super().update(instance, validated_data)
+            old_hostname = instance.resolved_middleware.get("hostname")
+            updated_instance: Asset = super().update(instance, validated_data)
             cache.delete(f"asset:{instance.external_id}")
+            new_hostname = updated_instance.resolved_middleware.get("hostname")
+            response = push_config_to_middleware(
```
Comment (vigneshhari) [inline care/facility/api/serializers/asset.py]: Move to background task

File: care/facility/api/viewsets/asset.py [hunk 3]
```diff
@@ -413,6 +424,72 @@ def operate_assets(self, request, *args, **kwargs):
...
+            .filter(
+                current_location__facility=self.request.user.facility,
+                asset_class__in=[
+                    AssetClasses.ONVIF.name,
+                    AssetClasses.HL7MONITOR.name,
+                ],
+            )
+            .annotate(
+                resolved_middleware_hostname=Coalesce(
+                    NullIf(
```
Comment (vigneshhari) [inline care/facility/api/viewsets/asset.py:467]: See if you can use the or statement here

File: care/facility/migrations/0426_dailyround_is_parsed_by_ocr_and_more.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+            ),
+        ),
+        migrations.AlterField(
+            model_name="facility",
+            name="middleware_address",
+            field=models.CharField(
+                db_index=True, default=None, max_length=200, null=True
+            ),
+        ),
+        migrations.AddIndex(
```
Comment (vigneshhari) [inline care/facility/migrations/0426_dailyround_is_parsed_by_ocr_and_more.py]: Remove Index

**Review by vigneshhari — CHANGES_REQUESTED** [review]

File: care/facility/api/serializers/asset.py (hunk 1 above)
Comment (sainak) [reply care/facility/api/serializers/asset.py]: Converting this to or was breaking the validation

File: care/facility/api/viewsets/asset.py (hunk 3 above)
Comment (sainak) [reply care/facility/api/viewsets/asset.py:467]: Converting this to or was returning assets from other hostnames as well

**Pull request merged**